classes/task/robot_cleanup.php

```diff
@@ -0,0 +1,66 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+
+/**
+ * local_linkchecker_robot
+ *
+ * @package    local_linkchecker_robot
+ * @copyright  2016 Brendan Heywood <brendan@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace local_linkchecker_robot\task;
+
+/**
+ * robot_cleanup
+ *
+ * @package    local_linkchecker_robot
+ * @copyright  2016 Brendan Heywood <brendan@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class robot_cleanup extends \core\task\scheduled_task {
+
+    /**
+     * Get task name
+     */
+    public function get_name() {
+        return get_string('robotcleanup', 'local_linkchecker_robot');
+    }
+
+    /**
+     * Execute task
+     *
+     * Finds and deletes all bad URLs which belong to previous scrape session, but not the current one.
+     */
+    public function execute() {
+        global $DB;
+        $currenttime = time();
+        $retentionperiod = \local_linkchecker_robot\robot\crawler::get_config()->retentionperiod;
+        $lastcrawlend = \local_linkchecker_robot\robot\crawler::get_config()->crawlend;
+        if ($retentionperiod) {
+            $param = array(
+                'currenttime' => $currenttime,
+                'lastcrawlfinished' => $lastcrawlend,
+                'retentionwise' => $currenttime - $retentionperiod
+            );
+            $where = 'lastcrawled <= :currenttime AND lastcrawled <= :lastcrawlfinished AND lastcrawled <= :retentionwise';
```

Comment:
It would be nicer if this wasn't one long line, so maybe wrap it similar to the sql which each AND on the next line and lined up